AWSAppSyncClient/AWSAppSyncHTTPNetworkTransport.swift

```diff
@@ -8,42 +8,87 @@ import Foundation
 import AWSCore
 
 public class AWSAppSyncHTTPNetworkTransport: AWSNetworkTransport {
-    let url: URL
-    let session: URLSession
-    var region: AWSRegionType? = nil
-    let serializationFormat = JSONSerializationFormat.self
-    var credentialsProvider: AWSCredentialsProvider? = nil
-    var apiKeyAuthProvider: AWSAPIKeyAuthProvider? = nil
-    var userPoolsAuthProvider: AWSCognitoUserPoolsAuthProvider? = nil
-    var oidcAuthProvider: AWSOIDCAuthProvider? = nil
-    var endpoint: AWSEndpoint? = nil
-    let authType: AWSAppSyncAuthType
-    var retryStrategy: AWSAppSyncRetryStrategy
-    var activeTimers: [String: DispatchSourceTimer] = [:]
-    
-    /// Creates a network transport with the specified server URL and session configuration.
+    public enum AppSyncAuthProvider {
```

Comment:
curious what do we gain by having this under `AWSAppSyncHTTPNetworkTransport` instead of it's own file?